src/Microsoft.AspNetCore.TestHost/WebSocketClient.cs

```diff
@@ -181,12 +190,20 @@ bool IHttpWebSocketFeature.IsWebSocketRequest
 
             Task<WebSocket> IHttpWebSocketFeature.AcceptAsync(WebSocketAcceptContext context)
             {
-                Context.HttpContext.Response.StatusCode = 101; // Switching Protocols
-
                 var websockets = TestWebSocket.CreatePair(context.SubProtocol);
-                _clientWebSocketTcs.SetResult(websockets.Item1);
-                _serverWebSocket = websockets.Item2;
-                return Task.FromResult<WebSocket>(_serverWebSocket);
+                if (_clientWebSocketTcs.TrySetResult(websockets.Item1))
+                {
+                    Context.HttpContext.Response.StatusCode = 101; // Switching Protocols
```

Comment:
I'll do the same for the 500.

Would you also like me to fix the `IDE0001 Name can be simplified` information message?

`returnTask.FromResult(_serverWebSocket)` instead of  
`return Task.FromResult<WebSocket>(_serverWebSocket)`  
as the result type is an exact match with the declared task type.